### PR TITLE
docs: adds storybook to angular

### PR DESCRIPTION
**Pull request by hippee-lee**
# Important
I'm opening this b/c of the build issues observed in the original: https://github.com/vmware/clarity/pull/4646

It builds and has all of the storybook change. 

## Run Storybook in Multiple tabs
```
1. $ npm ci (in root directory and all packages)
2. cd packages/angular
3. $ npm run start:dev
4. OPEN A NEW TAB
5. $ npm run storybook
6. OPEN A NEW TAB
7. $ npm run watch:scss
```
## Run Storybook in one tab

```
1. $ npm ci (in root directory and all packages)
2. cd into packages/angular
3. $ npm run start:storybook
```
You will need to be familiar with tmux keychords
See [here](https://github.com/rse/stmux#usage) for usage info.

There are some optimizations to be made here so that there is only one command to do **all the things** needed to build and run storybook (stmux, node threading or rust for perfomance if that ever becomes an issue).

Dark theme isn't working so that will be an additional commit along with a more comprehensive audit for things I didn't catch yet that are left over from getting things up and running.

Signed-off-by: Matt Hippely <mhippely@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?
Adds Storybook for Angular components. 
There are some POCs and the rest are stub files for the components. 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
No storybook for angular

Issue Number: N/A

## What is the new behavior?
Story book for angular. 
## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/angular/.storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc.
+ * All Rights ReserveThis software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import '@clr/icons';
+import { addDecorator, addParameters } from '@storybook/angular';
+import { withCssResources } from '@storybook/addon-cssresources';
+import darkTheme from '../dist/clr-ui/clr-ui-dark.min.css';
```
Comment (Jinnie) [inline packages/angular/.storybook/preview.js:10]: I think this should be `../../clr-ui/clr-ui-dark.min.css`
`npm run build-storybook` fails for me with 
```
Error: Can't resolve '../dist/clr-ui/clr-ui-dark.min.css'
```
I guess it's related.

**Comment by Jinnie** [comment]
Why do we have so many dependencies on clarity core?
I shouldn't need to build core to be able to run the ng-storybook. Or I'm missing something?

File: packages/angular/.storybook/preview.js (hunk 1 above)
Comment (hippee-lee) [reply packages/angular/.storybook/preview.js:10]: Thanks for pointing this out. I hadn't given the build command any work yet. 
I updated the build-command to handle building the css before it builds storybook, like it does when run for development.

**Comment by hippee-lee** [comment]
@Jinnie - wrt core are you referring to the tsconfig paths or something else?
I put those there with the expectation that we will start to use core components in the angular ones when we start using them in angular land.

**Comment by Jinnie** [comment]
tsconfig and several storybook-related script commands in package.json
I see. I just needed to explicitly build core for them, though we don't yet really need them, so I decided to report.

**Comment by hippee-lee** [comment]
for tsconfig and the copy scripts = I added them since I already needed icons and beleive that we will need them soon enough in angular. 

I'd rather keep them but am not strongly tied to that preference.

File: packages/angular/.storybook/TODO.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Storybook For Clarity Angular
+
+## Broken things
+
+1. Dark theme is included as [Object object]
+2. ~~Troubleshoot which add on is breaking the storybook build w/ @clr/angular~~
+3. ~~icons not showing in demo code~~
+4. ~~Figma add-on seems to have issues with dark theme scale (component is very small)~~
```
Comment (gnomeontherun) [inline packages/angular/.storybook/TODO.md:8]: I prefer we don't keep this file.

Comment (hippee-lee) [reply packages/angular/.storybook/TODO.md:8]: I'll delete it before squash and merge.

**Pull request merged**